modules/oe_content_featured_media_field/tests/src/FunctionalJavascript/FeaturedMediaEntityBrowserWidgetTest.php

```diff
@@ -0,0 +1,253 @@
+    $this->getSession()->getPage()->checkField('entity_browser_select[media:1]');
+    $this->getSession()->getPage()->pressButton('Select image');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+
+    // Assert the image was selected and the widget shows the proper buttons.
+    $this->assertSession()->pageTextContains('Image 1');
+    $this->assertSession()->buttonNotExists('Select images');
+    $this->assertSession()->buttonExists('Remove');
+    $this->assertSession()->buttonExists('Edit');
+
+    // Add the other media image item.
+    $this->getSession()->getPage()->pressButton('Add another item');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    $this->getSession()->getPage()->pressButton('Select images');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    $this->getSession()->switchToIFrame('entity_browser_iframe_test_images');
+    $this->getSession()->getPage()->checkField('entity_browser_select[media:2]');
+    $this->getSession()->getPage()->pressButton('Select image');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    $this->assertSession()->pageTextContains('Image 2');
+    $this->assertSession()->buttonNotExists('Select images');
+
+    // Check that 'Image 1' media item is placed before 'Image 2'.
+    $this->assertOrderInPage(['Image 1', 'Image 2']);
+
+    // Fill in the other fields and save the node.
+    $this->getSession()->getPage()->fillField('featured_media_field[0][caption]', 'Image 1 caption');
+    $this->getSession()->getPage()->fillField('featured_media_field[1][caption]', 'Image 2 caption');
+    $this->getSession()->getPage()->fillField('Title', 'Test entity browser widget');
+    $this->getSession()->getPage()->pressButton('Save');
+
+    // Assert that the values were saved.
+    $this->assertSession()->pageTextContains('Featured media field');
+    $this->assertSession()->pageTextContains('Image 1');
+    $this->assertSession()->pageTextContains('Image 2');
+    $this->assertSession()->pageTextContains('Image 1 caption');
+    $this->assertSession()->pageTextContains('Image 2 caption');
+    // Assert items order is the same as it was in the form.
+    $this->assertOrderInPage(['Image 1', 'Image 2']);
+
+    // Edit the node to reorder field items.
+    $this->drupalGet('node/1/edit');
+    file_put_contents('test.html', $this->getSession()->getPage()->getOuterHtml());
+    $dragged = $this->getSession()->getPage()->find('css', 'table#featured-media-field-values > tbody > tr:nth-child(1) a.tabledrag-handle');
+    $target = $this->getSession()->getPage()->find('css', 'table#featured-media-field-values > tbody > tr:nth-child(2) a.tabledrag-handle');
+    $dragged->dragTo($target);
+    $this->assertSession()->assertWaitOnAjaxRequest();
+
+    // Check that 'Image 1' media item is placed after 'Image 2'.
+    $this->assertOrderInPage(['Image 2', 'Image 1']);
+    $this->getSession()->getPage()->pressButton('Save');
+
+    // Assert items order was saved.
```

Comment:
Again, assert the save message to be sure there was no error. And after, load the node and assert correct values in storage.